ocaml/libs/tracing/tracing.mli

```diff
@@ -0,0 +1,112 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+(* Basic types *)
+
+module SpanKind : sig
+  type t = Server | Consumer | Client | Producer | Internal
+
+  val to_string : t -> string
+end
+
+module SpanContext : sig
+  type t
+
+  val to_traceparent : t -> string
+
+  val of_traceparent : string -> t option
+end
+
+module Span : sig
+  type t
+
+  val get_span_context : t -> SpanContext.t
+
+  val set_span_kind : t -> SpanKind.t -> unit
+end
+
+module Tracer : sig
+  type t
+
+  val span_of_span_context : SpanContext.t -> string -> Span.t
+
+  val start :
+       ?kind:SpanKind.t
+    -> tracer:t
+    -> name:string
+    -> parent:Span.t option
+    -> unit
+    -> (Span.t option, exn) result
+
+  val finish : Span.t option -> (unit, exn) result
+
+  val assert_finished : Span.t option -> bool
+end
+
+module TracerProvider : sig
+  type t
+
+  val find_tracer : provider:t -> name:string -> Tracer.t
+end
+
+module TracerProviders : sig
+  val find_or_create_tracer :
+    provider:TracerProvider.t -> name:string -> Tracer.t
+
+  val set_default :
+       tags:(string * string) list
+    -> endpoints:string list
+    -> processors:string list
+    -> filters:string list
+    -> enabled:bool
+    -> unit
+
+  val set :
+       ?status:bool
+    -> ?tags:(string * string) list
+    -> ?endpoints:string list
+    -> ?filters:string list
+    -> ?processors:string list
+    -> name_label:string
+    -> unit
+    -> unit
+
+  val get_default : unit -> (TracerProvider.t, exn) result
+
+  val get_default_tracer : name:string -> Tracer.t
+end
+
+type blob = Span.t
+
+type t = blob option
+
+val empty : t
+
+val null : t -> bool
+
+val t_of_string : string -> t
+
+val string_of_t : t -> string
+
+val json_of_t : t -> string option
```

Comment:
It is not obvious why this ever can be None. Is JSON's Null not the right value?

Reply:
Because t can be None, would `{}` be a better default? I think we could condense to_string and to_json.

Reply:
I would use JSON's `null`